src/notification.cpp

```diff
@@ -1186,18 +1200,62 @@ void Notification::close()
     user activates the notification, which may occur long after the notification is
     published.
  */
+/*!
+    \qmlsignal Notification::actionReplyInvoked(string name, string replyText)
+
+    Emitted when a notification action that requires reply text is activated by the user. \a name indicates the name of the invoked action. \a replyText contains the user text.
+
+    Handling the \c actionReplyInvoked signal is only effective if the process is running when the
+    user activates the notification, which may occur long after the notification is
+    published.
+ */
+/*!
+    \fn void Notification::actionReplyInvoked(const QString &name, const QString &replyText)
+
+    Emitted when a notification action that requires reply text is activated by the user. \a name indicates the name of the invoked action. \a replyText contains the user text.
+
+    Handling the \c actionReplyInvoked signal is only effective if the process is running when the
+    user activates the notification, which may occur long after the notification is
+    published.
+ */
 void Notification::checkActionInvoked(uint id, QString actionKey)
 {
     Q_D(Notification);
     if (id == d->replacesId) {
-        emit actionInvoked(actionKey);
-
-        if (actionKey == DEFAULT_ACTION_NAME) {
-            emit clicked();
+        // TODO: check reply is filled in for actions that require them
+        foreach (const QVariant &action, d->remoteActions) {
+            QVariantMap vm = action.value<QVariantMap>();
+            const QString actionName = vm["name"].value<QString>();
+            if (!actionName.isEmpty() && actionName == actionKey) {
+                if (vm.contains("reply")) { // Need reply
+                    const QVariantMap reply = vm["reply"].value<QVariantMap>();
+                    if (d->replyText.isEmpty() // Reply is empty
+                        || (reply.contains("choices") && !reply["choices"].value<QStringList>().contains(d->replyText) // and not a valid choice 
+                            && (!reply.contains("editable") || reply["editable"].toBool() == false))) { // and not editable
+                        // TODO: Some sort of signal of rejection to the sender?
+                        return;
+                    }
+                    emit actionReplyInvoked(actionKey, d->replyText);
+                } else {
+                    emit actionInvoked(actionKey);
+                }
+                if (actionKey == DEFAULT_ACTION_NAME) {
+                    emit clicked();
```

Comment:
Pondering if this inside the foreach could break some cases where the "default" action gets signaled even if the notification doesn't explicitly include it.